airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -225,7 +225,31 @@ def __init__(self,  # pylint: disable=too-many-arguments,too-many-locals
         self.pod_template_file = pod_template_file
         self.name = self._set_name(name)
 
-    def execute(self, context):
+    @staticmethod
+    def create_labels_for_pod(context):
+        """
+        Generate labels for the pod s.t. we can track it in case of Operator crash
+
+        :param context:
+        :return:
+        """
+        labels = {
+            'dag_id': context['dag'].dag_id,
+            'task_id': context['task'].task_id,
+            'exec_date': context['ts'],
+            'try_number': context['ti'].try_number,
+        }
+        # In the case of sub dags this is just useful
+        if context['dag'].parent_dag:
```

Comment:
Would `dag.is_subdag` be a better check ?

```suggestion
        if context['dag'].is_subdag:
```